<launch>

<arg name="robot" default="$(env ROBOT)"/>
<arg name="hold_time" default="8.0"/>
<arg name="resolution" default="0.01"/>
<arg name="turn_speed" 		default="0.3"/>
<arg name="drive_speed" 	default="0.2"/>
<arg name="angle_tolerance" 	default="0.5"/>
<arg name="distance_tolerance" 	default="0.10"/>
<arg name="show_cells_"         default="false"/>

<group ns="$(arg robot)">
    <param name="tf_prefix" value="$(arg robot)"/>

    <!-- Minimal -->
    <include file="$(find turtlebot_bringup)/launch/minimal.launch">
    </include>
    <param name="mobile_base/base_frame" value="$(arg robot)/base_footprint"/>
    <param name="mobile_base/odom_frame" value="$(arg robot)/odom"/>

    <!-- Communication node -->
    <node pkg="turtlebot_deployment" type="inter_robot_communication" name="communication">
        <param name="robot_name" value="$(arg robot)"/>
    </node>

    <!-- Deployment node -->
    <node pkg="turtlebot_deployment" type="simple_deployment" name="deployment" output="screen">
        <param name="robot_name" 	value="$(arg robot)"/>
        <param name="hold_time" 	value="$(arg hold_time)"/>
        <param name="resolution" 	value="$(arg resolution)"/>
	<param name="show_cells_" 	value="$(arg show_cells_)"/>
    </node>  

    <!-- Turtlebot teleop -->
    <node pkg="turtlebot_teleop" type="turtlebot_teleop_key" name="turtlebot_teleop_keyboard"  output="screen">
    	<param name="scale_linear" value="0.5" type="double"/>
    	<param name="scale_angular" value="1.5" type="double"/>
    	<remap from="turtlebot_teleop_keyboard/cmd_vel" to="cmd_vel_mux/input/teleop"/>
    </node>

</group>

</launch>
